### ISSUE TITLE
Buffering on fast connections on Nvidia Shield

### ISSUE THREAD

**Issue by juanma-cvega** [faush01/plugin.video.embycon#103]
Hi,
I have a really weird issue when using the addon on my 300Mbps connection on my Nvidia Shield. I have constant buffering on any video file, it doesn't matter whether it's 4k or FHD, a movie or a TV show. The thing I do to fix this is either connect to a VPN or, the weirder one, watch another video file (or the same) from another Kodi in my network. The other option is to connect via wifi where I get around a 100Mbps. Or connect through an outlet in the house connected to the router with a cable that gives me max 100Mbps connection. 
It seems like the speed of the connection is actually breaking the stream. This is the only addon I have problems with and when using the Emby android app it works perfectly so it doesn't seem like an Emby problem.
I tried looking into the logs but I couldn't see any thing that could point to the problem. Here is a link to a log file with Embycon debug logs enabled. 
https://paste.ubuntu.com/p/fF9GNfqH6K/

**Comment by faush01** [comment]
I see a lot of this
CRenderManager::WaitForBuffer - timeout waiting for buffer
in your log.
One thing you could try is playing with the Kodi buffering settings in the advancedsettings.xml settings

https://kodi.wiki/view/HOW-TO:Modify_the_video_cache

**Comment by juanma-cvega** [comment]
I've done that already. I've set it to 300MB as recommended for a 3GB device like the Shield. But it's weird that I only have this problem when on a fast connection. When using a VPN, even though is not full speed, it's quite higher than what I need to limit it to to avoid buffering. It's as if leaving the connection to the Emby server to the VPN fixes the problem. 
I've also tried changing settings in the network connection like setting a static IP or using the Google DNS. Currently, I've set up a QoS on the wifi connection for the Shield to less than 100Mbps to make it work. Any time I set it up to a higher value the buffering starts. The same thing doesn't happen on PC so it seems like an Android TV/Shield problem.

**Comment by faush01** [comment]
try setting it to no buffering at all, 3 I think it is, I found on a fast system I had that turning off all buffering helped with seeking and startup plus felt like it was more stable.

**Comment by juanma-cvega** [comment]
I've tried that as well, same result. I think the buffering message in this case is not that it's waiting for the buffering to complete but that it cannot buffer at all. It's not able to download the stream or maybe there is something that makes it download it really slowly. Even with the buffer mode to not buffer I still get buffering. I can change the QoS with a phone app and it's almost instantaneous. When the QoS is enabled, it just works. The moment it's disabled, I get buffering. And if I reenable it while a video is buffering, it just starts working fine after a couple of seconds.

**Comment by faush01** [comment]
You could try the latest beta, I have made a few changes to the way playback works, I am not sure if it will help in your situation but it is worth a try.

**Comment by juanma-cvega** [comment]
You mean the auto play branch? I just tested it but it still has the same problem. I also tested on an Android phone (an old OnePlus 2) and I have the same problem so it seems like the problem is related to Android. I tried on 4G and I didn't have the problem but I don't get more than 30Mbps that way. I'll see if I can try on another wifi connection this weekend just in case it's related to the connection.

**Comment by faush01** [comment]
Yes the auto play branch is currently in beta, the latest beta 1.9.93 is from that branch

**Comment by juanma-cvega** [comment]
I've just tested the new speed limit you added to the branch version set to 6MB and to 13MB with a FullHD film but it still has the same issue. I've also tried on my android phone and on another wifi from a different ISP with the same results.

**Comment by faush01** [comment]
To test this use the latest beta.
I will need a log file with EmbyCon debug logging turned on in the addon settings. Make sure you restart kodi after enabling the debug settings.
It should be up to the server now, the log will confirm that

**Comment by juanma-cvega** [comment]
I'm not so sure this is the network causing the problem. The android Emby app works fine with no buffering wired. Granted it seems more android related than anything so maybe the problem requires a special treatment for android devices. I'm going to try to reach the Emby forum to see if they have any idea of what's different with android.

**Comment by faush01** [comment]
I just don't know, not even sure what to try next.

**Comment by juanma-cvega** [comment]
I just checked the forum and from the threads I guess you would be member TeamB? So I guess if I'm the only one asking about this it may be something related to my setup. 
I'm going to try to first installed Kodi 19 nightly on my phone and see if I get any different result. I'm also going to try on a vanilla setup without any other addons. 
Do you know of any free Emby server I can connect to to try the addon? So I can discard problems with this particular server.

**Comment by faush01** [comment]
I don't know of a free emby server.
You could also try directly playing the file from a file share to see what happens.

**Comment by juanma-cvega** [comment]
I just tried with Kodi 19 nightly and the matrix branch but I hit this error
```
2020-07-01 14:18:12.361 T:4021    ERROR <general>: GetDirectory - Error getting plugin://plugin.video.embycon/?content_type=video
2020-07-01 14:18:12.370 T:3952    ERROR <general>: CGUIMediaWindow::GetDirectory(plugin://plugin.video.embycon/?content_type=video) failed
2020-07-01 14:18:19.819 T:4032    ERROR <general>: Exception in thread Thread-3:
                                                   Traceback (most recent call last):
                                                     File "/data/user/0/org.xbmc.kodi/cache/apk/assets/python3.7/lib/python3.7/threading.py", line 926, in _bootstrap_inner
                                                       self.run()
                                                     File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/plugin.video.embycon/resources/lib/websocket_client.py", line 262, in run
                                                       setting_result = json_rpc('Settings.getSettingValue').execute(params)
                                                   NameError: name 'json_rpc' is not defined
                                                   
2020-07-01 14:18:19.821 T:4032    ERROR <general>: 
```

Any idea?

**Comment by faush01** [comment]
looks like a bug, will fix when I get a chance, kodi 19 is not the priority at the moment.

**Comment by juanma-cvega** [comment]
Ok, I'll wait for that to run the test then. Do you want me to open a different issue to keep track of the problem?

**Comment by faush01** [comment]
just post it in the forum.

**Comment by juanma-cvega** [comment]
done

**Comment by juanma-cvega** [comment]
I was able to test the latest version on Kodi Matrix but I still see the problem. I also don't know what else to test. I don't think this is related to the network as it works on other devices like PC or Mac so seems more like an Android thing. I still don't get though how I am the only one posting about this problem so either it really has something to do with my setup (even though I doubt it as I've tested a vanilla setup with just the addon and on a different network) or I'm the only Android user out there that has bothered to post the problem. In any case feel free to close the ticket if you want as I'm out of ideas so not sure whether you want to have it open for reference.
